### ISSUE TITLE
Environmental Issues

### ISSUE THREAD

**Issue by DoThingYo** [hu-zijing/B2-DiffuRL#4]
Hi Zijing,

May I ask how to install the running environments?

**Comment by hu-zijing** [comment]
Hello! 

We haven't had a chance to tidy up the environment yet. The following are the packages included in our environment. But it contains many **unnecessary packages** because we also use this environment for other projects. Hope this can help you :-).

~~~
- absl-py==2.0.0
- accelerate==0.25.0
- aiohttp==3.9.1
- aiosignal==1.3.1
- annotated-types==0.6.0
- antlr4-python3-runtime==4.9.3
- anyio==4.2.0
- appdirs==1.4.4
- async-timeout==4.0.3
- attrs==23.1.0
- bert-score==0.3.13
- bitsandbytes==0.41.3.post2
- blobfile==2.0.2
- braceexpand==0.1.7
- certifi==2023.7.22
- charset-normalizer==3.2.0
- click==8.1.7
- clip==0.2.0
- cmake==3.27.2
- contextlib2==21.6.0
- contourpy==1.1.0
- cycler==0.11.0
- datasets==3.5.0
- decord==0.6.0
- diffusers==0.24.0
- dill==0.3.7
- distro==1.9.0
- dm-tree==0.1.8
- docker-pycreds==0.4.0
- et-xmlfile==1.1.0
- exceptiongroup==1.2.0
- fairscale==0.4.13
- filelock==3.12.2
- fonttools==4.42.1
- frozenlist==1.4.1
- fsspec==2023.12.2
- ftfy==6.1.3
- gitdb==4.0.11
- gitpython==3.1.40
- h11==0.14.0
- httpcore==1.0.2
- httpx==0.26.0
- huggingface-hub==0.19.4
- idna==3.4
- image-reward==1.5
- imageio==2.33.1
- importlib-metadata==7.0.0
- inflect==7.0.0
- iopath==0.1.10
- jinja2==3.1.2
- joblib==1.3.2
- kaleido==0.2.1
- kiwisolver==1.4.5
- lazy-loader==0.3
- lightning-utilities==0.10.0
- lit==16.0.6
- lxml==4.9.3
- markupsafe==2.1.3
- matplotlib==3.7.2
- ml-collections==0.1.1
- mpmath==1.3.0
- multidict==6.0.4
- multiprocess==0.70.15
- networkx==3.1
- numpy==1.25.2
- nvidia-cublas-cu11==11.10.3.66
- nvidia-cuda-cupti-cu11==11.7.101
- nvidia-cuda-nvrtc-cu11==11.7.99
- nvidia-cuda-runtime-cu11==11.7.99
- nvidia-cudnn-cu11==8.5.0.96
- nvidia-cufft-cu11==10.9.0.58
- nvidia-curand-cu11==10.2.10.91
- nvidia-cusolver-cu11==11.4.0.1
- nvidia-cusparse-cu11==11.7.4.91
- nvidia-nccl-cu11==2.14.3
- nvidia-nvtx-cu11==11.7.91
- omegaconf==2.3.0
- open-clip-torch==2.24.0
- openai==1.8.0
- opencv-python==4.8.1.78
- openpyxl==3.1.2
- packaging==23.1
- pandas==2.0.3
- peft==0.7.1
- pillow==10.0.0
- plotly==5.24.1
- portalocker==2.8.2
- progressbar2==4.3.2
- protobuf==4.25.1
- psutil==5.9.6
- pyarrow==19.0.1
- pycryptodomex==3.18.0
- pydantic==2.5.2
- pydantic-core==2.14.5
- pyparsing==3.0.9
- python-dateutil==2.8.2
- python-utils==3.8.1
- pytorch-lightning==2.1.2
- pytz==2023.3
- pyyaml==6.0.1
- regex==2023.8.8
- requests==2.32.3
- safetensors==0.4.1
- scikit-image==0.22.0
- scikit-learn==1.3.2
- scipy==1.11.3
- sentencepiece==0.1.99
- sentry-sdk==1.39.1
- setproctitle==1.3.3
- six==1.16.0
- smmap==5.0.1
- sniffio==1.3.0
- svgwrite==1.4.3
- sympy==1.12
- tenacity==9.0.0
- threadpoolctl==3.2.0
- tifffile==2023.12.9
- timm==0.6.13
- tokenizers==0.15.0
- torch==2.0.1
- torchmetrics==1.2.1
- torchvision==0.15.2
- tqdm==4.67.1
- transformers==4.36.1
- triton==2.0.0
- typing-extensions==4.7.1
- tzdata==2023.3
- urllib3==2.0.4
- visual-genome==1.1.1
- wandb==0.16.1
- wcwidth==0.2.13
- webdataset==0.2.86
- xxhash==3.5.0
- yarl==1.9.4
- zipp==3.17.0
~~~

**Comment by DoThingYo** [comment]
Hi Zijing,

Thanks so much for your reply!

I tried these packages. But the following errors indicate that the version of huggingface-hub is not appropriate. Could you please have a check of this? I'm on a ubuntu server.

INFO: pip is looking at multiple versions of datasets to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 2), -r requirements.txt (line 23) and huggingface-hub==0.19.4 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested huggingface-hub==0.19.4
    accelerate 0.25.0 depends on huggingface-hub
    datasets 3.5.0 depends on huggingface-hub>=0.24.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip to attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

**Comment by HaiyanggJiang** [comment]
openAI的triton是不能通过pip intsall安装

**Comment by DoThingYo** [comment]
> openAI的triton是不能通过pip intsall安装

Thank you for your reply!

I have changed the version of the packages, and it works now. Triton is installed through pip. But I indeed got something wrong when running the code. I will start a new issue.
